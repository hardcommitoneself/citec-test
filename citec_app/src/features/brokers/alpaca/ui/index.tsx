// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { useState } from 'react'
import { Spinner } from 'reactstrap'
import { useDispatch } from 'react-redux'
import { ProviderAlpaca } from 'features/brokers/alpaca/utils/provider-alpaca'
import {
   setBrokerConnected,
   setA1c_t,
   setAccountType,
   setAccountTypeSelect,
   resetBrokers,
} from 'store/slices/brokers'
import { AlpacaApi } from 'features/brokers/alpaca/utils/hooks/api/alpaca-api'

export let requestIsValid: any

export const AlpacaOauth = () => {
   const dispatch = useDispatch()

   const [error, setError] = useState('')
   const [loading, setLoading] = useState(false)

   async function connectToAlpaca() {
      setLoading(true)
      const random_string = '4Frt9x6'
      const codeURI =
         `https://app.alpaca.markets/oauth/authorize?` +
         `response_type=code&` +
         `client_id=${process.env.REACT_APP_ALPACA_CLIENT_ID}&` +
         `redirect_uri=${process.env.REACT_APP_ALPACA_REDIRECT_URI}&` +
         `state=${random_string}&` +
         `scope=account:write%20trading`
      try {
         const alpacaResponse: any = await openAlpacaPopUp(codeURI)
         if (!requestIsValid(random_string, alpacaResponse.state)) {
            throw new Error('Alpaca Authentication Invalid')
         }

         const tokenAlpaca = await ProviderAlpaca.SetTokenAuth0Alpaca(
            alpacaResponse.code
         )
         if (tokenAlpaca.code == 'sucess') {
            dispatch(setBrokerConnected({ brokerConnected: 'Alpaca' }))
            dispatch(setA1c_t({ a1c_t: tokenAlpaca.value }))
            getAccountTypes()
            setLoading(false)
         } else {
            throw new Error(tokenAlpaca.value)
         }
      } catch (error) {
         let message = ''
         if (error instanceof Error) {
            message = error.message
         } else {
            message = String(error)
         }

         setError(message)
         setLoading(false)
      }
   }

   requestIsValid = (inital_state: string, returned_state: string) => {
      return inital_state === returned_state
   }

   const getAccountTypes = () => {
      AlpacaApi({
         type: 'LIVE',
         url: 'account',
         method: 'GET',
      })
         .then(function (response) {
            if (response.data.status != 'PAPER_ONLY') {
               dispatch(setAccountType({ accountType: ['PAPER', 'LIVE'] }))
            } else {
               dispatch(setAccountType({ accountType: ['PAPER'] }))
            }

            dispatch(setAccountTypeSelect({ accountTypeSelect: 'PAPER' }))
         })
         .catch(function (error) {
            if (error.response) {
               if (error.response.status == '403') {
                  dispatch(resetBrokers())
                  localStorage.removeItem('a1c_t')
               } else {
                  console.log(error.response.data)
                  console.log(error.response.status)
                  console.log(error.response.headers)
               }
            }
         })
   }

   const openAlpacaPopUp = (uri: string) => {
      return new Promise((resolve, reject) => {
         const authWindow = window.open(
            uri,
            '_blank',
            'toolbar=yes,scrollbars=yes,resizable=yes,top=400,left=400,width=500,height=400'
         )
         let snippet: any = ''

         const interval = setInterval(async () => {
            try {
               if (authWindow == null || authWindow.closed) {
                  reject('An ocurred error')
                  clearInterval(interval)
               } else {
                  snippet =
                     authWindow && authWindow.location && authWindow.location.search
               }
            } catch (e) {
               console.log((e as Error).message)
            }
            if (snippet) {
               const rawCode = snippet.substring(1)
               const code = JSON.parse(
                  '{"' + rawCode.replace(/&/g, '","').replace(/=/g, '":"') + '"}',
                  function (key, value) {
                     return key === '' ? value : decodeURIComponent(value)
                  }
               )
               authWindow?.close()
               resolve(code)
               clearInterval(interval)
            }
         }, 100)
      })
   }

   return (
      <>
         {loading ? (
            <Spinner animation="border" role="status">
               <span className="sr-only">Loading...</span>
            </Spinner>
         ) : (
            <div
               className="account-item-body"
               data-testid="lnk-alpaca-oauth"
               onClick={() => connectToAlpaca()}
            >
               <img src={require('assets/img/brokers/alpaca.png')} />
               Alpaca
            </div>
         )}
         {error && <p>{error}</p>}
      </>
   )
}
