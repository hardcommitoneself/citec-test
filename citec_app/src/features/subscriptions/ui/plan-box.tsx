// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import React from 'react'
const tick = require('assets/img/tick_list.png')

const PlanBox = ({ plan, price, list, button, price_id }: any) => {

   const handleSubmit = async (price_id: string) => {

      if (price_id == 'Custom') {
         window.location = "mailto:info@citecsolutions.com" as never
      } else if (price_id == 'Coming-soon') {
         window.location.href = "/subscriptions"
      }
      const token = localStorage.getItem('token')

      try {
         const checkoutSession = await fetch(process.env.REACT_APP_API_URL + "subscriptions/checkout/?price_id=" + price_id, {
            method: "POST",
            headers: {
               'Authorization': 'Bearer ' + token,
            },
         }).then((res) => res.json())
         const checkoutObject = JSON.parse(checkoutSession);
         window.location.assign(checkoutObject.url)
      } catch (err) {
         console.log(err);
      }
   }

   return (
      <div className={`PricingBox ${plan == 'Custom' && 'disable'}`}>
         <h1 className="heading">{plan}</h1>
         {price == 'Custom' ? (
            <h2 className="price CustomPrice ">{price}</h2>
         ) : (
            <h2 className="price">
               ${price}
               <span>.99</span>
            </h2>
         )}
         <button onClick={() => handleSubmit(price_id)}>{button}</button>

         <ul>
            {list.map((Each: boolean | React.ReactChild | React.ReactFragment | React.ReactPortal | null | undefined) => (
               <li>
                  <img src={tick} alt="" />
                  <p>{Each}</p>
               </li>
            ))}
         </ul>
      </div>
   )
}
export default PlanBox
