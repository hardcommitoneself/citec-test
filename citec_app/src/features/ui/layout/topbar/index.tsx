// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { AppLink } from 'routes/AppLink'
import { Icon } from 'assets/icons'
import { Link } from 'react-router-dom'
import { TopbarUserMenu } from './topbar-user-menu'

const search = require('assets/img/search.png')

interface TopbarProps {
   color?: 'white' | 'green'
}

export const Topbar = ({ color = 'white' }: TopbarProps) => {
   return (
      <div
         data-testid="topbar"
         className={`dashbaord_header px-7  ${
            color === 'green' ? 'bg-[#0baca9] h-[80px] ' : 'h-[100px] '
         }`}
      >
         {color === 'green' ? (
            <AppLink to="/">
               <Icon.LogoCT />
            </AppLink>
         ) : (
            <>
               <div />
               <div className="search_wrapper !hidden">
                  <img src={search} alt="" />
                  <input type="text" placeholder="Search  anything..." />
               </div>
            </>
         )}

         <div
            className={`right_side ${
               color === 'white' ? 'text-[#79828D]' : 'text-white'
            }`}
         >
            <nav>
               <Link to="/" className="hidden">
                  <Icon.Message color={color} />
               </Link>
               <Link to="/">
                  <Icon.Home color={color} />
               </Link>
            </nav>

            <TopbarUserMenu />
         </div>
      </div>
   )
}
