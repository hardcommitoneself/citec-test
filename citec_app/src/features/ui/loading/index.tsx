// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const Loading = () => {
   const generateBars = () => {
      const bars = []

      for (let index = 0; index < 10; index++) {
         bars.push(<div key={index} className={`rect${index + 1}`}></div>)
      }

      return bars
   }

   return (
      <div className="flex flex-col justify-center items-center h-screen bg-black w-full z-50">
         <div className="spinner" data-testid="loading">
            {generateBars()}
         </div>

         <h3 id="loading-content" className="mt-3 text-white"></h3>
      </div>
   )
}
