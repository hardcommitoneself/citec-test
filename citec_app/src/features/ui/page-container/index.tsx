// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Alert, AlertTitle } from '@mui/material'
import { FC, ReactNode, useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { useAppSelector } from 'store/hooks'
import { resetError } from 'store/slices/errors'

interface PageContainerProps {
   actionBtn?: ReactNode
   bgWhite?: boolean
   className?: string
   secondaryTitle?: string
   titlePage?: string
   error?: string
   alertTitle?: any
}

export const PageContainer: FC<PageContainerProps> = ({
   actionBtn,
   bgWhite,
   children,
   className,
   error,
   secondaryTitle,
   titlePage,
   alertTitle,
}) => {
   const errorOnRedux = useAppSelector((state) => state.errors)
   const dispatch = useDispatch()

   const Group = require('assets/img/group.png')

   useEffect(() => {
      return () => {
         if (errorOnRedux) dispatch(resetError())
      }
   }, [])

   return (
      <div
         className={`${className} my-8 mx-10 ${
            bgWhite && 'bg-white p-4'
         } rounded-2xl relative`}
         id="pagecontainer"
      >
         <div className="criteria_title flex justify-between">
            {titlePage && (
               <h3 className="mb-4 font-semibold text-[26px] text-[#2d405a]">
                  {titlePage}
                  {alertTitle && (
                     <img
                        className="moreinfo ml-8"
                        src={Group}
                        alt=""
                        onClick={() => alertTitle()}
                     />
                  )}
               </h3>
            )}
            {actionBtn && <div>{actionBtn}</div>}
         </div>

         {secondaryTitle && (
            <h3 className="mt-16 mb-5 text-neutral-500 text-2xl text-center">
               {secondaryTitle}
            </h3>
         )}

         {error ? (
            <Alert severity="error" className="mb-3">
               <AlertTitle>
                  <strong>Error</strong>
               </AlertTitle>
               {error}
            </Alert>
         ) : (
            children
         )}
      </div>
   )
}
