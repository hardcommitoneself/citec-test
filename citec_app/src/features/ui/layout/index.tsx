// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Outlet, useLocation } from 'react-router-dom'
import Swal from 'sweetalert2'

import { AppNavigate, Paths } from 'routes'
import { Topbar } from './topbar'
import {
   useValidateSoldPosition,
   useValidateTickers,
   useValidationPorfolioLoaded,
} from 'features/utils/hooks'
import Leftbar from './leftbar/leftbar-container'
import { useAppSelector } from 'store/hooks'

interface LayoutProps {
   hiddenLeftbar?: boolean
}

export const Layout = ({ hiddenLeftbar }: LayoutProps) => {
   const { pathname } = useLocation()
   const workflow = useAppSelector((state) => state.workflow.type)
   const tickers = useAppSelector((state) => state.portfolio.ticker)

   const { isPortfolioLoaded, pagesThatNoNeedPortofolioLoaded } =
      useValidationPorfolioLoaded()

   const { pagesThatNeedSoldPosition, isSomeSellPosition, is10PercentSold } =
      useValidateSoldPosition()

   const { isSomeInvalidTicker, pagesThatNoNeedTickerValidation } =
      useValidateTickers(tickers)

   const workflowWithSoldValidation = ['load', 'rebalance']

   let to = ''

   const renderPopup = (errorMsg: string) => {
      Swal.fire({
         icon: 'error',
         title: 'Oops...',
         text: errorMsg,
      })
   }

   if (
      isPortfolioLoaded &&
      isSomeInvalidTicker &&
      !pagesThatNoNeedTickerValidation.includes(pathname as Paths)
   ) {
      renderPopup('Please review the tickets in your portfolio')
      to = '/portfolio-loader'
   }

   if (
      !isPortfolioLoaded &&
      !pagesThatNoNeedPortofolioLoaded.includes(pathname as Paths)
   ) {
      renderPopup('You need to upload a portfolio')

      to = '/workflow'
      return <AppNavigate to={to as Paths} />
   }

   if (workflowWithSoldValidation.includes(workflow)) {
      if (
         pagesThatNeedSoldPosition.includes(pathname as Paths) &&
         !is10PercentSold
      ) {
         renderPopup('You need to sell at least 10% of the portfolio to continue')
         to = '/tax-harvesting'
      }

      if (
         pagesThatNeedSoldPosition.includes(pathname as Paths) &&
         !isSomeSellPosition
      ) {
         renderPopup('You need to sell some position')

         to = '/tax-harvesting'
      }
   }

   let classNameContent = 'container-with-left'
   if (hiddenLeftbar) classNameContent = 'w-full'

   if (to && workflowWithSoldValidation.includes(workflow))
      return <AppNavigate to={to as Paths} />

   return (
      <div className="min-h-screen flex" data-testid="bars">
         <Leftbar hidden={hiddenLeftbar} />
         <div className={classNameContent}>
            <Topbar color={hiddenLeftbar ? 'green' : 'white'} />

            <Outlet />
         </div>
      </div>
   )
}
