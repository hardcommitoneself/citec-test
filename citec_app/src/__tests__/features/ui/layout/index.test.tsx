// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { Layout } from 'features/ui/layout'
import { exampleReduxState } from 'features/utils/const/exampleReduxState'

describe('test Bars component', () => {
   // intercept the useAuth0 function and mock it
   jest.mock('@auth0/auth0-react', () => ({
      Auth0Provider: ({ children }: { children: JSX.Element }) => children,
      withAuthenticationRequired: (component: any) => component,
      useAuth0: () => {
         return {
            isLoading: false,
            user: { sub: 'foobar' },
            isAuthenticated: true,
            loginWithRedirect: jest.fn(),
         }
      },
   }))

   const mockStore = configureStore()
   const store = mockStore(exampleReduxState)

   test('Render <Bars />', () => {
      const { getByTestId, container } = render(
         <Provider store={store}>
            <MemoryRouter>
               <Layout />
            </MemoryRouter>
         </Provider>
      )
      expect(getByTestId('bars')).toBeTruthy()
      expect(container).toMatchSnapshot()
   })
})
