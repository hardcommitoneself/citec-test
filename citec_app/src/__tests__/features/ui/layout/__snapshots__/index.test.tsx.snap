// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Bars component Render <Bars /> 1`] = `
<div>
  <div
    class="min-h-screen flex"
    data-testid="bars"
  >
    <div
      class="Sidebar false undefined"
      data-testid="leftbar"
      id="active"
    >
      <div
        class="top_area"
      >
        <a
          href="/"
        >
          <img
            alt=""
            class="logo"
            src="dashboard_logo.png"
          />
        </a>
        <span
          class="bars"
        >
          <img
            alt=""
            src="dashboard_bars.png"
          />
        </span>
      </div>
      <nav>
        <a
          aria-current="page"
          class="link not active"
          data-testid="leftbar-link"
          href="/"
        >
          <svg
            fill="currentColor"
            height="23"
            viewBox="0 0 23 23"
            width="23"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.7835 0.819572C12.6812 0.716454 12.5595 0.634607 12.4254 0.578752C12.2914 0.522897 12.1476 0.494141 12.0023 0.494141C11.8571 0.494141 11.7133 0.522897 11.5792 0.578752C11.4452 0.634607 11.3235 0.716454 11.2212 0.819572L1.31957 10.7212C1.21645 10.8235 1.13461 10.9452 1.07875 11.0792C1.0229 11.2133 0.994141 11.3571 0.994141 11.5023C0.994141 11.6476 1.0229 11.7914 1.07875 11.9254C1.13461 12.0595 1.21645 12.1812 1.31957 12.2835C1.42237 12.3854 1.54429 12.4661 1.67833 12.5208C1.81238 12.5756 1.95591 12.6033 2.1007 12.6025H3.20088V20.3038C3.20088 20.8874 3.4327 21.447 3.84535 21.8597C4.258 22.2723 4.81767 22.5041 5.40124 22.5041H18.6034C19.187 22.5041 19.7467 22.2723 20.1593 21.8597C20.5719 21.447 20.8038 20.8874 20.8038 20.3038V12.6025H21.904C22.1957 12.6025 22.4756 12.4866 22.6819 12.2803C22.8882 12.074 23.0041 11.7941 23.0041 11.5023C23.005 11.3575 22.9772 11.214 22.9225 11.08C22.8677 10.9459 22.787 10.824 22.6851 10.7212L12.7835 0.819572ZM5.40124 20.3038V9.75304L12.0023 3.15196L18.6034 9.75304V20.3038H5.40124Z"
            />
          </svg>
           
          <span
            class="ml-3"
          >
            Home
          </span>
        </a>
        <a
          class="link not"
          data-testid="leftbar-link"
          href="/workflow"
        >
          <svg
            fill="currentColor"
            height="23"
            viewBox="0 0 26 26"
            width="23"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.3208 0.813477L0.25 6.60554V9.61888H25.25V6.58955L12.3208 0.813477ZM23.5833 7.95789H1.91667V7.64926L12.3459 2.64492L23.5833 7.66525V7.95789ZM0.25 25.8135H25.25V20.4153H0.25V25.8135ZM1.91667 22.0763H23.5833V24.1525H1.91667V22.0763ZM3.16667 10.8646H4.83333V19.1695H3.16667V10.8646ZM20.6667 10.8646H22.3333V19.1695H20.6667V10.8646ZM9 10.8646H10.6667V19.1695H9V10.8646ZM14.8333 10.8646H16.5V19.1695H14.8333V10.8646Z"
            />
          </svg>
           
          <span
            class="ml-3"
          >
            Portfolio Builder
          </span>
        </a>
        <a
          class="link not"
          data-testid="leftbar-link"
          href="/universe-screener"
        >
          <svg
            fill="currentColor"
            height="22"
            viewBox="0 0 18 18"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.5 18H7.5C7.10218 18 6.72064 17.842 6.43934 17.5607C6.15804 17.2794 6 16.8978 6 16.5V10.8075L0.4425 5.25C0.160809 4.96999 0.00167459 4.58968 0 4.1925V1.5C0 1.10218 0.158035 0.720644 0.43934 0.43934C0.720644 0.158035 1.10218 0 1.5 0H16.5C16.8978 0 17.2794 0.158035 17.5607 0.43934C17.842 0.720644 18 1.10218 18 1.5V4.1925C17.9983 4.58968 17.8392 4.96999 17.5575 5.25L12 10.8075V16.5C12 16.8978 11.842 17.2794 11.5607 17.5607C11.2794 17.842 10.8978 18 10.5 18ZM1.5 1.5V4.1925L7.5 10.1925V16.5H10.5V10.1925L16.5 4.1925V1.5H1.5Z"
            />
          </svg>
           
          <span
            class="ml-3"
          >
            Screener
          </span>
        </a>
      </nav>
    </div>
    <div
      class="container-with-left"
    >
      <div
        class="dashbaord_header px-7  h-[100px] "
        data-testid="topbar"
      >
        <div />
        <div
          class="search_wrapper !hidden"
        >
          <img
            alt=""
            src="search.png"
          />
          <input
            placeholder="Search  anything..."
            type="text"
          />
        </div>
        <div
          class="right_side text-[#79828D]"
        >
          <nav>
            <a
              class="hidden"
              href="/"
            >
              <svg
                fill="none"
                height="23"
                viewBox="0 0 22 23"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.4167 22.4993H15.5834C15.3403 22.4993 15.1071 22.4028 14.9352 22.2309C14.7633 22.059 14.6667 21.8258 14.6667 21.5827C14.6667 21.3396 14.7633 21.1064 14.9352 20.9345C15.1071 20.7626 15.3403 20.666 15.5834 20.666H17.4167C18.1461 20.666 18.8456 20.3763 19.3613 19.8606C19.877 19.3448 20.1667 18.6454 20.1667 17.916V16.0827C20.1667 15.8396 20.2633 15.6064 20.4352 15.4345C20.6071 15.2626 20.8403 15.166 21.0834 15.166C21.3265 15.166 21.5597 15.2626 21.7316 15.4345C21.9035 15.6064 22.0001 15.8396 22.0001 16.0827V17.916C21.9986 19.1311 21.5153 20.2961 20.656 21.1553C19.7968 22.0145 18.6319 22.4979 17.4167 22.4993Z"
                  fill="#79828D"
                />
                <path
                  d="M0.916666 7.83333C0.673551 7.83333 0.440394 7.73675 0.268485 7.56485C0.0965771 7.39294 0 7.15978 0 6.91666V5.08333C0.00145554 3.8682 0.484808 2.70326 1.34403 1.84403C2.20326 0.984808 3.3682 0.501456 4.58333 0.5L6.41666 0.5C6.65978 0.5 6.89294 0.596577 7.06485 0.768485C7.23675 0.940394 7.33333 1.17355 7.33333 1.41667C7.33333 1.65978 7.23675 1.89294 7.06485 2.06485C6.89294 2.23676 6.65978 2.33333 6.41666 2.33333H4.58333C3.85399 2.33333 3.15451 2.62306 2.63879 3.13879C2.12306 3.65451 1.83333 4.35399 1.83333 5.08333V6.91666C1.83333 7.15978 1.73676 7.39294 1.56485 7.56485C1.39294 7.73675 1.15978 7.83333 0.916666 7.83333Z"
                  fill="#79828D"
                />
                <path
                  d="M6.41666 22.4993H4.58333C3.3682 22.4979 2.20326 22.0145 1.34403 21.1553C0.484808 20.2961 0.00145554 19.1311 0 17.916L0 16.0827C0 15.8396 0.0965771 15.6064 0.268485 15.4345C0.440394 15.2626 0.673551 15.166 0.916666 15.166C1.15978 15.166 1.39294 15.2626 1.56485 15.4345C1.73676 15.6064 1.83333 15.8396 1.83333 16.0827V17.916C1.83333 18.6454 2.12306 19.3448 2.63879 19.8606C3.15451 20.3763 3.85399 20.666 4.58333 20.666H6.41666C6.65978 20.666 6.89294 20.7626 7.06485 20.9345C7.23675 21.1064 7.33333 21.3396 7.33333 21.5827C7.33333 21.8258 7.23675 22.059 7.06485 22.2309C6.89294 22.4028 6.65978 22.4993 6.41666 22.4993Z"
                  fill="#79828D"
                />
                <path
                  d="M21.0834 7.83333C20.8403 7.83333 20.6071 7.73675 20.4352 7.56485C20.2633 7.39294 20.1667 7.15978 20.1667 6.91666V5.08333C20.1667 4.35399 19.877 3.65451 19.3613 3.13879C18.8456 2.62306 18.1461 2.33333 17.4167 2.33333H15.5834C15.3403 2.33333 15.1071 2.23676 14.9352 2.06485C14.7633 1.89294 14.6667 1.65978 14.6667 1.41667C14.6667 1.17355 14.7633 0.940394 14.9352 0.768485C15.1071 0.596577 15.3403 0.5 15.5834 0.5H17.4167C18.6319 0.501456 19.7968 0.984808 20.656 1.84403C21.5153 2.70326 21.9986 3.8682 22.0001 5.08333V6.91666C22.0001 7.15978 21.9035 7.39294 21.7316 7.56485C21.5597 7.73675 21.3265 7.83333 21.0834 7.83333Z"
                  fill="#79828D"
                />
              </svg>
            </a>
            <a
              href="/"
            >
              <svg
                fill="#79828D"
                height="23"
                viewBox="0 0 23 23"
                width="23"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.7835 0.819572C12.6812 0.716454 12.5595 0.634607 12.4254 0.578752C12.2914 0.522897 12.1476 0.494141 12.0023 0.494141C11.8571 0.494141 11.7133 0.522897 11.5792 0.578752C11.4452 0.634607 11.3235 0.716454 11.2212 0.819572L1.31957 10.7212C1.21645 10.8235 1.13461 10.9452 1.07875 11.0792C1.0229 11.2133 0.994141 11.3571 0.994141 11.5023C0.994141 11.6476 1.0229 11.7914 1.07875 11.9254C1.13461 12.0595 1.21645 12.1812 1.31957 12.2835C1.42237 12.3854 1.54429 12.4661 1.67833 12.5208C1.81238 12.5756 1.95591 12.6033 2.1007 12.6025H3.20088V20.3038C3.20088 20.8874 3.4327 21.447 3.84535 21.8597C4.258 22.2723 4.81767 22.5041 5.40124 22.5041H18.6034C19.187 22.5041 19.7467 22.2723 20.1593 21.8597C20.5719 21.447 20.8038 20.8874 20.8038 20.3038V12.6025H21.904C22.1957 12.6025 22.4756 12.4866 22.6819 12.2803C22.8882 12.074 23.0041 11.7941 23.0041 11.5023C23.005 11.3575 22.9772 11.214 22.9225 11.08C22.8677 10.9459 22.787 10.824 22.6851 10.7212L12.7835 0.819572ZM5.40124 20.3038V9.75304L12.0023 3.15196L18.6034 9.75304V20.3038H5.40124Z"
                />
              </svg>
            </a>
          </nav>
          <div
            class="profile_area"
            data-testid="topbar-user-menu"
          >
            <p />
            <img
              alt=""
              class="profile_icon"
            />
            <div
              class="dropdown"
            >
              <span
                aria-expanded="false"
                aria-haspopup="true"
                class="cursor-pointer dropdown-toggle"
                data-toggle="dropdown"
              >
                 
              </span>
              <div
                aria-hidden="true"
                class="dropdown-menu dropdown-menu-right"
                role="menu"
                tabindex="-1"
              >
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <a
                    class="flex"
                    href="/profile"
                  >
                    My Profile
                  </a>
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  Logout
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
